Common/Orders/OrderTypes.cs

```diff
@@ -139,4 +173,47 @@ public enum OrderStatus {
         Invalid
     }
 
+
+    /// <summary>
+    /// OrderStatus extensions
+    /// </summary>
+    public static class OrderStatusEx
+    {
+        /// <summary>
+        /// Test for open orders
+        /// </summary>
+        /// <param name="source">Order status</param>
+        /// <returns>True if open status</returns>
+        public static bool IsOpen(this OrderStatus source)
+        {
+            return source == OrderStatus.New
+                || source == OrderStatus.Submitted;
+        }
+
+        /// <summary>
+        /// Test for Canceled, Filled, Invalid or PartiallyFilled orders.
+        /// </summary>
+        /// <param name="source">Order status</param>
+        /// <returns>True if completed status</returns>
+        public static bool IsCompleted(this OrderStatus source)
+        {
+            return source == OrderStatus.Canceled
+                || source == OrderStatus.Filled
+                || source == OrderStatus.Invalid
+                || source == OrderStatus.PartiallyFilled;
```

Comment:
I'm a little confused as to the difference in the method naming between `IsCompleted` and `IsFinal`. When I read `IsCompleted` it makes me think the same thing as `IsFinal`. Also, when I read `IsCompleted` it makes me think the order is completely finished, i.e., not partially filled, but completely filled or canceled/invalid.

Can you describe the reasoning for both? As it is currently defined, `IsCompleted` looks to be more of an `IsUpdateable`?

Also, maybe these two can be merged into a single `IsCompleted` that is the same definition as `IsFinal`?